src/opencensus/proto/metrics/v1/metrics.proto

```diff
@@ -195,26 +197,37 @@ message DistributionValue {
 
   // A Distribution may optionally contain a histogram of the values in the
   // population. The bucket boundaries for that histogram are described by
-  // bucket_bounds. This defines size(bucket_bounds) + 1 (= N)
-  // buckets. The boundaries for bucket index i are:
-  //
-  // (-infinity, bucket_bounds[i]) for i == 0
-  // [bucket_bounds[i-1], bucket_bounds[i]) for 0 < i < N-2
-  // [bucket_bounds[i-1], +infinity) for i == N-1
+  // BucketOptions.
   //
-  // i.e. an underflow bucket (number 0), zero or more finite buckets (1
-  // through N - 2, and an overflow bucket (N - 1), with inclusive lower
-  // bounds and exclusive upper bounds.
-  //
-  // If bucket_bounds has no elements (zero size), then there is no
-  // histogram associated with the Distribution. If bucket_bounds has only
-  // one element, there are no finite buckets, and that single element is the
-  // common boundary of the overflow and underflow buckets. The values must
-  // be monotonically increasing.
-  //
-  // Don't change bucket boundaries within a timeseries if your backend
-  // doesn't support this.
-  repeated double bucket_bounds = 4;
+  // If bucket_options has no type, then there is no histogram associated with
+  // the Distribution.
+  message BucketOptions {
+    oneof type {
+      // Bucket with explicit bounds.
+      Explicit explict = 1;
+    }
+
+    // Specifies a set of buckets with arbitrary upper-bounds.
+    // This defines size(bounds) + 1 (= N) buckets. The boundaries for bucket
+    // index i are:
+    //
+    // [0, bucket_bounds[i]) for i == 0
+    // [bucket_bounds[i-1], bucket_bounds[i]) for 0 < i < N-1
+    // [bucket_bounds[i-1], +infinity) for i == N-1
+    message Explicit {
+      // The values must be monotonically increasing and > 0.
```

Comment:
+ cannot repeat?

Reply:
[Wikipedia](https://en.wikipedia.org/wiki/Monotonic_function) has me worried with "a function that increases monotonically does not exclusively have to increase, it simply must not decrease"

How about "strictly increasing" ?